boshio/boshio.go

```diff
@@ -194,7 +196,10 @@ func (c Client) retryableRequest(stemcellURL string, byteRange string) ([]byte,
 	byteRangeHeader := fmt.Sprintf("bytes=%s", byteRange)
 	req.Header.Add("Range", byteRangeHeader)
 
-Retry:
+	return c.retry(req)
+}
+
+func (c Client) retry(req *http.Request) ([]byte, error) {
```

Comment:
One of the reasons we opted for a second function was to ensure the `defer Body.Close()` were run before the next request was started. We could switch this to a `for` loop instead but we have a slight preference for the recursive call. Your call.